### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description

Updating Basic Auth docs with information about new configuration options

### CODE REVIEW

File: docs/content/guides/security/auth/extauth/basic_auth/_index.md [hunk 1]
```diff
@@ -206,6 +206,61 @@ If the auth config has been received successfully, you should see the log line:
 "logger":"extauth","caller":"runner/run.go:179","msg":"got new config"
 ```
 
+### Extended configuration
+{{% notice warning %}}
+The auth configuration format shown on this page was introduced with **Gloo Enterprise**, release [**1.16.0**]({{ .Site.Data.Solo.DocsVersion }}/reference/changelog/enterprise).
```
Comment (sheidkamp) [inline docs/content/guides/security/auth/extauth/basic_auth/_index.md:211]: I want to get the link to render like it does from the `extauth_version_info_note` file, but can't figure out how.

File: changelog/v1.16.0-beta20/sha1-basic-auth-docs.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+changelog:
```
Comment (sheidkamp) [inline changelog/v1.16.0-beta20/sha1-basic-auth-docs.yaml:1]: Comment to hold auto-merging until docs reviews.

File: docs/content/guides/security/auth/extauth/basic_auth/_index.md [hunk 3]
```diff
@@ -206,6 +206,67 @@ If the auth config has been received successfully, you should see the log line:
...
```
Comment (Nadine2016) [inline docs/content/guides/security/auth/extauth/basic_auth/_index.md:211]:
```suggestion
The auth configuration format that is shown on this page was introduced with [Gloo Edge Enterprise release 1.16.0]({{< versioned_link_path fromRoot="/reference/changelog/enterprise" >}}).
```

File: docs/content/guides/security/auth/extauth/basic_auth/_index.md [hunk 4]
```diff
@@ -206,6 +206,67 @@ If the auth config has been received successfully, you should see the log line:
...
+If you are using an earlier version, please refer to [this page]({{ .Site.Data.Solo.DocsVersion }}/guides/security/auth/extauth/configuration_format_history/)
```
Comment (Nadine2016) [inline docs/content/guides/security/auth/extauth/basic_auth/_index.md:212]:
```suggestion
If you use an earlier version, refer to [this page]({{< versioned_link_path fromRoot="/guides/security/auth/extauth/configuration_format_history/" >}}) to find the auth configuration format that you must use. 
```

File: docs/content/guides/security/auth/extauth/basic_auth/_index.md [hunk 5]
```diff
@@ -206,6 +206,67 @@ If the auth config has been received successfully, you should see the log line:
...
+to see which configuration formats are supported by each version.
```
Comment (Nadine2016) [inline docs/content/guides/security/auth/extauth/basic_auth/_index.md:213]:
```suggestion
```

File: docs/content/guides/security/auth/extauth/basic_auth/_index.md (hunk 1 above)
Comment (sheidkamp) [reply docs/content/guides/security/auth/extauth/basic_auth/_index.md]: @Nadine2016 - do you know how to get that formatting to work?

File: docs/content/guides/security/auth/extauth/basic_auth/_index.md [hunk 6]
```diff
@@ -206,6 +206,67 @@ If the auth config has been received successfully, you should see the log line:
...
+{{% /notice %}}
+
+An extended configuration is available that allows use of the SHA1 hashing algorithm instead of APR.
+
+This configuration splits definition the users from the definition of the encryption algorithm. The configuration with the same functionality as the example would look like:
```
Comment (Nadine2016) [inline docs/content/guides/security/auth/extauth/basic_auth/_index.md:217]: doesn't the SHA1 example also split the definition of users from the definition of encryption algorithms? if so, maybe change that sentence to: 

The following configuration defines a list of users, and the salt and hashedPassword that they need to use to authenticate successfully. The salt and hashed password are encrypted by using the APR algorithm.

File: docs/content/guides/security/auth/extauth/basic_auth/_index.md [hunk 7]
```diff
@@ -206,6 +206,67 @@ If the auth config has been received successfully, you should see the log line:
...
+      encryption:
+        apr: {}
+      userList:
+        users:
+          user:
+            salt: "TYiryv0/"
+            hashedPassword: "8BvzLUO9IfGPGGsPnAgSu1"
+{{< /highlight >}}
+
+To switch to SHA1 encryption, the following config would support the same user list (with updated salted and hashed passwords):
```
Comment (Nadine2016) [inline docs/content/guides/security/auth/extauth/basic_auth/_index.md:237]: You can change the encryption algorithm for the salt and hashed password to SHA1 as shown in the following example.

**Review by Nadine2016 — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Comment by sam-heilbron** [comment]
comment to trigger bulldozer

**Pull request merged**